emmet-core/emmet/core/phonon.py

```diff
@@ -1,131 +1,816 @@
+                    + 1.0j * table[f"{col_prefix}eigendisplacements_imag"].to_numpy()[0]
+                )
+            elif k == "qpoint_labels":
+                config["labels_dict"] = dict(
+                    zip(v, table[f"{col_prefix}qpoint_labelled_points"].to_pylist()[0])
+                )
+            else:
+                config[k] = v
+        return cls(**config)
+
+    @classmethod
+    def from_phonopy(cls, phonon_bandstructure_file: str | Path):
+        """Create a PhononBS from phonopy .yaml output."""
+        with zopen(phonon_bandstructure_file, "rt") as f:
+            phonopy_bandstructure = yaml.safe_load(f.read())
+
+        phonopy_bandstructure.update(
+            qpoints=[entry["q-position"] for entry in phonopy_bandstructure["phonon"]],
+            frequencies=[
+                [branch["frequency"] for branch in entry["band"]]
+                for entry in phonopy_bandstructure["phonon"]
+            ],
+        )
+        return cls(**phonopy_bandstructure)
+
+
+class SumRuleChecks(BaseModel):
+    """Container class for defining sum rule checks."""
+
+    asr: float | None = Field(
+        None, description="The violation of the acoustic sum rule."
+    )
+    cnsr: float | None = Field(
+        None, description="The violation of the charge neutral sum rule."
+    )
+
+
+class PhononBSDOSTask(StructureMetadata):
+    """Phonon band structures and density of states data."""
+
+    identifier: str | None = Field(
+        None, description="The identifier of this phonon analysis task."
+    )
+
+    phonon_method: PhononMethod | None = Field(
+        None, description="The method used to calculate phonon properties."
     )
 
-    ph_bs: Optional[PhononBandStructureSymmLine] = Field(
+    phonon_bandstructure: PhononBS | None = Field(
         None,
         description="Phonon band structure object.",
     )
 
-    ph_dos: Optional[PhononDosObject] = Field(
+    phonon_dos: PhononDOS | None = Field(
         None,
         description="Phonon density of states object.",
     )
 
-    last_updated: Optional[datetime] = Field(
+    epsilon_static: Matrix3D | None = Field(
+        None, description="The high-frequency dielectric constant."
+    )
+
+    epsilon_electronic: Matrix3D | None = Field(
         None,
-        description="Timestamp for the most recent calculation for this Material document.",
+        description="The electronic contribution to the high-frequency dielectric constant.",
     )
 
-    # Make sure that the datetime field is properly formatted
-    @field_validator("last_updated", mode="before")
-    @classmethod
-    def handle_datetime(cls, v):
-        return convert_datetime(cls, v)
+    born: list[Matrix3D] | None = Field(
+        None,
+        description="Born charges, only for symmetrically inequivalent atoms",
+    )
 
+    # needed, e.g. to compute Grueneisen parameter etc
+    force_constants: list[list[Matrix3D]] | None = Field(
```

Comment:
`force_constants` are also outsourced to S3 in production I think. So should be optional?